docs/DeveloperGuide.adoc

```diff
@@ -840,18 +1097,71 @@ Use case ends.
 * 3a. The given index is invalid.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1. CRM Book shows an error message.
 +
 Use case resumes at step 2.
 
-_{More to be added}_
+=== Use case: Detect a duplicate Lead or Contact
+
+*MSS*
+
+1. User enters a Lead to add
+2. CRM Book determines if Lead can be added
+3. CRM Book displays message on whether Lead entered can be added or is a duplicate
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. User either entered Lead or add command incorrectly.
++
+[none]
+** 1a1. CRM displays error message.
++
+Use case ends.
+
+* 2a. Lead entered has a phone number or email address stored earlier.
+** 2a1. CRM displays message that a possible duplicate detected.
++
+Use case ends.
+
+* 2b. Lead entered does not have a phone number or email address stored earlier.
+** 2b1. CRM adds Lead.
+** 2b2. CRM displays message of successful addition of Lead.
++
+Use case ends.
+
+=== Use case: User seeks list of either Leads or Contacts
+
+*MSS*
+
+1. User enters command for list of either Lead or Contact.
+2. CRM retrieves Leads or Contacts.
+3. CRM displays list of Leads or Contacts.
+
+*Extensions*
+
+[none]
+* 1a. User enters command incorrectly.
+
+** 1a1. CRM displays error message.
++
+Use case ends.
++
+* 2a. Desired list is empty.
+** 2a1. CRM displays message that list is empty.
++
+Use case ends.
```

Comment:
Where are the other use cases for your enhancements? This is also part of documentation. @davidten @WoodyLau @zhuleyan @A0155428B